pkg/venafi/tpp/connector_test.go

```diff
@@ -530,44 +532,98 @@ func TestRequestCertificateWithValidHours(t *testing.T) {
+				{"/vedsdk/certificates/reset", `{"CertificateDN":"\\VED\\Policy\\application-team-1\\app1.example.com"}`,
+					"400 Bad request. Check the error in the response for details.", `{"Error":"Reset is not completed. No reset is required for the certificate."}`},
+				{"/vedsdk/certificates/request", `{"PolicyDN":"\\VED\\Policy\\application-team-1", "CASpecificAttributes":[{"Name":"Origin", "Value":"Venafi VCert-Go"}], "CertificateType":"AUTO", "DisableAutomaticRenewal":true, "KeyAlgorithm":"RSA", "Origin":"Venafi VCert-Go", "Reenable":true}`,
+					"200 OK", `{"CertificateDN":"\\VED\\Policy\\application-team-1\\app1.example.com","Guid":"{b86278bf-091f-4205-a0a3-8f3e8de140c3}"}`},
+			},
+		}, {
+			name: "should request even if reset fails with an unexpected 400",
+			mockCalls: []mockCall{
+				{"/vedsdk/certificates/reset", `{"CertificateDN":"\\VED\\Policy\\application-team-1\\app1.example.com"}`,
+					"400 Bad request. Check the error in the response for details.", `{"Error":"Invalid CertificateDN format. The Certificate DN contained null or white spaces for (CertificateDN)."}`},
+			},
+			expectErr: "while resetting certificate in case of previous enrollment: while resetting certificate due to a past failed enrollment. Status: 400 Bad request. Check the error in the response for details., Body: \r{\"Error\":\"Invalid CertificateDN format. The Certificate DN contained null or white spaces for (CertificateDN).\"}",
+		}, {
+			name: "should not request if reset fails with an unexpected 403",
+			mockCalls: []mockCall{
+				{"/vedsdk/certificates/reset", `{"CertificateDN":"\\VED\\Policy\\application-team-1\\app1.example.com"}`,
+					"403 Forbidden", `{"error":"insufficient_scope","error_description":"Grant rejected scope 'certificate:delete'. Call POST Authorize\/OAuth with the correct scope and restriction. Update the header with the new token and retry."}`},
+			},
+			expectErr: "while resetting certificate in case of previous enrollment: while resetting certificate due to a past failed enrollment. Status: 403 Forbidden, Body: \r{\"error\":\"insufficient_scope\",\"error_description\":\"Grant rejected scope 'certificate:delete'. Call POST Authorize\\/OAuth with the correct scope and restriction. Update the header with the new token and retry.\"}",
 		},
 	}
+
+	serverWith := func(t *testing.T, mockCalls []mockCall) (_ *httptest.Server, ca *x509.CertPool) {
+		callCount := new(int32)
+		server := httptest.NewTLSServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			i := atomic.AddInt32(callCount, 1) - 1
+			if i >= int32(len(mockCalls)) {
+				t.Fatalf("expected %d calls to %s but got %d", len(mockCalls), r.URL.Path, i+1)
+			}
+			assert.Equal(t, mockCalls[i].expectReqPath, r.URL.Path)
+			bytes, err := ioutil.ReadAll(r.Body)
+			require.NoError(t, err)
+			assert.JSONEq(t, mockCalls[i].expectReqBody, string(bytes))
+			writeRespWithCustomStatus(w, mockCalls[i].mockStatus, mockCalls[i].mockBody)
```

Comment:
The asserts in this function will be called from another goroutine, I think this might not be fully supported by go.

Reply:
`t.Log`, `t.Error`, `t.Fatal`, `t.Fail`, and `t.Fail` [are thread-safe](https://pkg.go.dev/testing):

> The other reporting methods, such as the variations of Log and Error, may be called simultaneously from multiple goroutines.

Since Testify rely on `fmt.Fprintf`, `t.Fail` and `t.FailNow`, it shouldn't pose a problem

Reply:
I was referring to these kind of messages that my IDE shows to me:
https://deepsource.io/directory/analyzers/go/issues/SCC-SA2002

Reply:
Looks like we cannot use FailNow and SkipNow:
- https://pkg.go.dev/testing#T.SkipNow
- https://pkg.go.dev/testing#T.FailNow

Reply:
Got it. It seems `t.FailNow` is the one that needs to be called from the test goroutine, not another goroutine, as per [this comment](https://github.com/stretchr/testify/issues/772#issuecomment-945166599):

> Using testify `assert` and `require` with goroutines is the same as using `t.Errorf()` and `t.FailNow()` respectively. `assert` doesn't document this but `require` [sort of does](https://pkg.go.dev/github.com/stretchr/testify@v1.7.0/require#hdr-Assertions). [`t.FailNow()`](https://pkg.go.dev/testing#T.FailNow) must be called from the goroutine running the test, I think this is the issue the OP is mentioning? [`t.Errorf()`](https://pkg.go.dev/testing#T.Errorf) does not have the same restriction. 
>
> The Go 1.16 release notes [also speak about this](https://golang.org/doc/go1.16#vet-testing-T) because calling `t.FailNow()` in a goroutine became a vet warning (use of the `require` package does not trigger this warning), they give the most simple example of how to solve it; just use `t.Error()`. Translating this to testify we would just say "use `assert` in goroutines, not `require`".

I'll make sure that only `assert` is used in the `httptest.Server`.